sql/information_schema/columns_table.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package information_schema
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+)
+
+// ColumnsNode wraps the information_schema.columns table as a way to resolve column defaults.
+type ColumnsNode struct {
+	plan.UnaryNode
+
+	// columnNameToColumn maps the name of a column (databaseName.tableName.columnName) with a non nil default value
+	// to its column object.
+	columnNameToColumn map[string]*sql.Column
+
+	// defaultToColumn maps the pointer of a sql.ColumnDefault value back to its original Column Object.
+	defaultToColumn map[*sql.ColumnDefaultValue]*sql.Column
+}
+
+var _ sql.Node = (*ColumnsNode)(nil)
```

Comment:
This needs to implement sql.Table in addition to sql.Node

Reply:
Won't this cause problems with resolution and bunch of analyzer logic? Essentially everywhere we are looking for a resolved table we are just getting a ColumnsTable. That's why I wanted this Node to wrap the ColumnTable

Reply:
Essentially we need to type the ColumnsTable in a ton of analyzer works. Wrapping a resolved table saves us a ton of potential edge cases IMO.